gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/workspace/HasBundlesDirListener.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.gradle.core.workspace;
+
+import org.eclipse.sapphire.FilteredListener;
+import org.eclipse.sapphire.PropertyContentEvent;
+
+/**
+ * @author Andy Wu
+ */
+public class HasBundlesDirListener extends FilteredListener<PropertyContentEvent>
```

Comment:
oh!  got it , sorry I was trying to make font-end logic fit my back-end logic , and I will fix my back-end logic and send new pull request cause it will be cheaper  to come to the same function .